docs/CodeLens.md

```diff
@@ -0,0 +1,56 @@
+
+## Code Lens
+
+### DTD features
+
+Code lens is provided in `.dtd` files to show where elements defined in the DTD are referenced.
+For instance, in the following code:
+
+```xml
+<!ELEMENT letter (to, from, title, body)>
+<!ELEMENT to (#PCDATA)>
+<!ELEMENT from (#PCDATA)>
+<!ELEMENT title (#PCDATA)>
+<!ELEMENT body (#PCDATA)>
+```
+
+There will be a code lens entry after the first line that points from the
+definition of the element `to` to where it is used in the definition of the
+element `letter`.
+
+Here is a demonstration of the above Code Lens:
+
+![A letter DTD element is declared, which consists of to, from title, and body elements. The definitions for these sub elements have a code lens above, which allows ofr navigating to the letter element declaration.](./images/code-lens/DTDElementCodeLens.gif)
```

Comment:
To be consistent, please rename folder `code-lens` to `Codelens`. The idea is to have the same name than page. I think we should follow the same rule for validation (rename the images/validation to images/Validation)

Reply:
Also applied this to Preferences